src/slam_toolbox_common.cpp

```diff
@@ -228,12 +228,15 @@ void SlamToolbox::publishTransformLoop(
   while (rclcpp::ok()) {
     {
       boost::mutex::scoped_lock lock(map_to_odom_mutex_);
-      geometry_msgs::msg::TransformStamped msg;
-      msg.transform = tf2::toMsg(map_to_odom_);
-      msg.child_frame_id = odom_frame_;
-      msg.header.frame_id = map_frame_;
-      msg.header.stamp = scan_timestamped + transform_timeout_;
-      tfB_->sendTransform(msg);
+      // Avoids publishing tf with initial 0.0 scan timestamp
+      if (scan_timestamped.seconds() > 0.0) {
```

Comment:
Also, check if the frame_id is `empty()`, that way we know for sure that there isn't a valid scan yet (time is `0` and frame is empty string)

Reply:
Good idea, made the change. To get the frame_id, I'm now storing the laser scan header instead of storing just the timestamp.